File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/job/JobCrafterDirectoryEntryRequestMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait JobCrafterDirectoryEntryRequestMessage extends MessageDeserializer {
  val messageId = 6043

  protected trait Message extends BaseMessage with Serializable {
    val messageId = JobCrafterDirectoryEntryRequestMessage.messageId

    val playerId: Int

    def serialize(buf: Buffer) {
      buf.writeInt(playerId)
    }

  }

  protected def create(playerId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val playerId = buf.readInt()

    create(playerId)
  }

}

object JobCrafterDirectoryEntryRequestMessage extends JobCrafterDirectoryEntryRequestMessage {
  case class MessageImpl(playerId: Int) extends Message
  type Target = MessageImpl

  override def create(playerId: Int) = MessageImpl(playerId)

  def apply(playerId: Int) = create(playerId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/JobCrafterDirectoryEntryRequestMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class JobCrafterDirectoryEntryRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6043;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int playerId;
    
    public JobCrafterDirectoryEntryRequestMessage() { }
    
    public JobCrafterDirectoryEntryRequestMessage(int playerId) {
        this.playerId = playerId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(playerId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        playerId = buf.readInt();
        if (playerId < 0)
            throw new RuntimeException("Forbidden value on playerId = " + playerId + ", it doesn't respect the following condition : playerId < 0");
    }
    
}
